File: 001_two_sum/twoSum.go
package main

import "fmt"

func main() {
	a := twoSum([]int{2, 3, 3}, 6)

	fmt.Printf("%v\n", a)
}

func twoSum(nums []int, target int) []int {
	var m = make(map[int]int)

	for k, v := range nums {
		left := target - v

		if val, ok := m[left]; ok {
			return []int{val, k}
		}

		m[v] = k
	}

	return []int{0, 0}
}

File: 001_two_sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const map = {};
  for(let i =0; i< nums.length; i++) {
    const item = nums[i]
    if(map[item] == undefined) {
      map[target - item] = i
    } else {
      return [map[item], i ]
    }
  } 
}

let a = twoSum([2,7,11,15], 9)
console.log('a', a)